src/components/missions.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import fetchMission from './API/apiMissions';
+import { reserveMission, leaveMission } from '../features/missionSclice';
+// header of the table
+function HeaderTable() {
+  return (
+    <table border={1} className="header_table">
+      <tr>
+        <th className="header_mission">Mission</th>
+        <th className="header_description">Description</th>
+        <th className="header_status">status</th>
+        <th className="header_action" aria-label="empty" />
+      </tr>
+    </table>
+
+  );
+}
+function JoinedMissions() {
+  const missions = useSelector((state) => state.missionReducer.missionStore);
+  const filteredMissions = missions.filter((mission) => mission.reserved);
+
+  return (
+    <div className="Joined_mission_container">
+
+      {filteredMissions.map((mission) => (
+        <div className="list_mission" key={mission.mission_id}>{mission.mission_name}</div>
+      ))}
+    </div>
+  );
+}
+
+export { JoinedMissions };
+// mission table
+function Mission({
+  id,
+  name,
+  description,
+  reserved,
+}) {
+  const dispatch = useDispatch();
+  return (
+    <div>
+      <table className="missions_table" border={1}>
+        <tr>
+          <td className="mission_name">{name}</td>
+          <td className="mission_description">{description}</td>
+          <td className="mission_status">
+            {reserved ? (
+              <button className="member_button" type="submit">Member</button>
+            ) : (
+              <button className="not_member_button" type="submit">Not a Member</button>
+            )}
+
+          </td>
+          <td className="mission_action">
+            {reserved ? (
+              <button type="submit" onClick={() => dispatch(leaveMission(id))} className="leave_button">Leave mission</button>
+            ) : (
+              <button type="submit" onClick={() => dispatch(reserveMission(id))} className="join_button">Join mission</button>
+
+            )}
+          </td>
+        </tr>
+      </table>
+    </div>
+  );
+}
+Mission.propTypes = { // Add
+  id: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
+  description: PropTypes.string.isRequired,
+  reserved: PropTypes.string.isRequired,
+};
+function Missions() {
+  const missionsArr = useSelector((state) => state.missionReducer.missionStore);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchMission());
+  }, [dispatch]);
```

Comment:
- [ ] You really did an amazing job. unlike the rockets, every time I join missions, navigate and come back to the mission page, the state is re-initialized and all my reservations are wiped out please kindly fix it